api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.expressions.Expression;
+
+/**
+ * An action that rewrites data files.
+ */
+public interface RewriteDataFiles extends
+    ConfigurableAction<RewriteDataFiles, RewriteDataFiles.Result>,
+    SnapshotUpdate<RewriteDataFiles, RewriteDataFiles.Result> {
+
+  /**
+   * Pass a row filter to filter {@link DataFile}s to be rewritten.
+   * <p>
+   * Note that all files that may contain data matching the filter may be rewritten.
+   *
+   * @param expr a row filter to filter out data files
+   * @return this for method chaining
+   */
+  RewriteDataFiles filter(Expression expr);
+
+  /**
+   * Enables or disables case sensitive expression binding.
+   *
+   * @param caseSensitive caseSensitive
+   * @return this for method chaining
+   */
+  RewriteDataFiles caseSensitive(boolean caseSensitive);
+
+  /**
+   * Pass a PartitionSpec id to specify which PartitionSpec should be used in DataFile rewrite
+   *
+   * @param specId PartitionSpec id to rewrite
+   * @return this for method chaining
+   */
+  RewriteDataFiles outputSpecId(int specId);
+
+  /**
+   * Specify the target rewrite data file size in bytes
```

Comment:
Mention that the default is the table's target size.

Reply:
Done.